### PR TITLE
Pin RuboCop to avoid failures due to renamed cops

### PR DESCRIPTION
**Pull request by pirj**
In RuboCop 1.0 this will stop happening, and we'll be able to relax it to:

    gem 'rubocop', '~> 1.0'

Caused trouble in #2215 and #2212

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -67,7 +67,7 @@ end
 
 gem "rubyzip", '~> 1.2'
 
-gem 'rubocop', "~> 0.74"
+gem 'rubocop', '~> 0.74.0'
```
Comment (JonRowe) [inline Gemfile]: Can we fix the failures instead and bump instead? It would be better to stay up to date?

Comment (pirj) [reply Gemfile]: Most of the failures come from `.rubocop_rspec_base.yml`. Should I update RuboCop for all repos?
I can do that as a quick follow-up, but what do you think of merging this as is not to block other pull requests? 

```
$ rubocop/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for LineLength.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for MethodLength.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for SpaceAroundEqualsInParameterDefault.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for FileName.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for IfUnlessModifier.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop.yml: Warning: no department given for PerlBackrefs.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for AccessModifierIndentation.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for Alias.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for CaseEquality.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for CollectionMethods.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for CyclomaticComplexity.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for Documentation.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for DotPosition.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for DoubleNegation.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for EachWithObject.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for FormatString.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for HashSyntax.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for Lambda.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for LineLength.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for BinaryOperatorParameterName.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for PercentLiteralDelimiters.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for PredicateName.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for Proc.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for RescueException.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for SignalException.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for SpaceAroundEqualsInParameterDefault.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for StringLiterals.
/Users/pirj/source/rspec-dev/repos/rspec-rails/.rubocop_rspec_base.yml: Warning: no department given for TrivialAccessors.
Error: The `Layout/AlignParameters` cop has been renamed to `Layout/ParameterAlignment`.
(obsolete configuration found in .rubocop_rspec_base.yml, please update it)
The `Layout/IndentAssignment` cop has been renamed to `Layout/AssignmentIndentation`.
(obsolete configuration found in .rubocop_rspec_base.yml, please update it)
The `Layout/IndentFirstArgument` cop has been renamed to `Layout/FirstArgumentIndentation`.
(obsolete configuration found in .rubocop_rspec_base.yml, please update it)
The `Layout/IndentFirstArrayElement` cop has been renamed to `Layout/FirstArrayElementIndentation`.
(obsolete configuration found in .rubocop_rspec_base.yml, please update it)
The `Layout/IndentHeredoc` cop has been renamed to `Layout/HeredocIndentation`.
(obsolete configuration found in .rubocop_rspec_base.yml, please update it)
```

Comment (JonRowe) [reply Gemfile]: I think rspec-rails is already divorced from the rest on this, given that its now independant from the others I think its ok to set its rubocop to newer on its own

Comment (pirj) [reply Gemfile]: Awesome!

Updated to latest 0.77.0.
Merged `.rubocop_rspec_base.yml` into `.rubocop.yml`.
Extracted those cops' configs that were marked as "Over time we'd like to get this down" to `.rubocop_todo.yml`.

File: .rubocop.yml [hunk 2]
```diff
@@ -1,72 +1,332 @@
...
+Style/IdenticalConditionalBranches:
+  Enabled: false
+
+Style/IfUnlessModifierOfIfUnless:
+  Enabled: false
+
+Style/IfInsideElse:
+  Enabled: false
+
+Style/IfUnlessModifier:
```
Comment (JonRowe) [inline .rubocop.yml]: Previously we disabled this, and we have a bunch of failures now due to it.

Comment (pirj) [reply .rubocop.yml]: I was misled by configuration composition.

**Comment by pirj** [comment]
`Style/Proc` was lost in translation, got it back to disabled state.
`rubocop lib` is green locally.

**Comment by pirj** [comment]
@JonRowe The build is green.

File: .rubocop.yml [hunk 3]
```diff
@@ -1,72 +1,333 @@
...
+
+Layout/ParameterAlignment:
+  EnforcedStyle: with_first_parameter
+
+Layout/SpaceInsidePercentLiteralDelimiters:
+  Enabled: false
 
 # We use spaces, so it's less of a change to stick with that.
-SpaceAroundEqualsInParameterDefault:
+Layout/SpaceAroundEqualsInParameterDefault:
```
Comment (JonRowe) [inline .rubocop.yml]: What makes me laugh here is the base file says the exact opposite 😂

File: .rubocop_todo.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Style/EachWithObject:
+  Enabled: false
+
+# ruby 1.8.7 only had support for hash rockets
+Style/HashSyntax:
+  EnforcedStyle: hash_rockets # default: ruby19
+
+# the new lambda syntax was not supported in ruby 1.8.7
+Style/Lambda:
+  Enabled: false
```
Comment (JonRowe) [inline .rubocop_todo.yml]: I'm going to assume the intent is to remove all these at some point?

**Comment by JonRowe** [comment]
Looks good, I'm going to merge this so we can fix builds etc. Comments are for conversation only, I also added back an explanation for a rule.

File: .rubocop_todo.yml (hunk 4 above)
Comment (pirj) [reply .rubocop_todo.yml]: Yep, extracted those with promises to improve things, or 1.8 support excuses.

**Pull request merged**